plasmapy_nei/nei/nei.py

```diff
@@ -1,7 +1,1363 @@
+        then ``time_start`` defaults to ``time_input[0]`` (if given) and
+        zero seconds otherwise.
+
+    time_max: astropy.units.Quantity
+        The maximum time for the simulation.  If density and/or
+        temperature are given by arrays, then this argument must be less
+        than ``time_input[-1]``.
+
+    max_steps: `int`
+        The maximum number of time steps to be taken during a
+        simulation.
+
+    dt: astropy.units.Quantity
+        The time step.  If ``adapt_dt`` is `False`, then ``dt`` is the
+        time step for the whole simulation.
+
+    dt_max: astropy.units.Quantity
+        The maximum time step to be used with an adaptive time step.
+
+    dt_min: astropy.units.Quantity
+        The minimum time step to be used with an adaptive time step.
+
+    adapt_dt: `bool`
+        If `True`, change the time step based on the characteristic
+        ionization and recombination time scales and change in
+        temperature.  Not yet implemented.
+
+    safety_factor: `float` or `int`
+        A multiplicative factor to multiply by the time step when
+        ``adapt_dt`` is `True`.  Lower values improve accuracy, whereas
+        higher values reduce computational time.  Not yet implemented.
+
+    tol: float
+        The absolute tolerance to be used in comparing ionic fractions.
+
+    verbose: bool, optional
+        A flag stating whether or not to print out information for every
+        time step. Setting ``verbose`` to `True` is useful for testing.
+        Defaults to `False`.
+
+    abundances: dict
+
+    Examples
+    --------
+
+    >>> import numpy as np
+    >>> import astropy.units as u
+
+    >>> inputs = {'H': [0.9, 0.1], 'He': [0.9, 0.099, 0.001]}
+    >>> abund = {'H': 1, 'He': 0.085}
+    >>> n = np.array([1e9, 1e8]) * u.cm ** -3
+    >>> T_e = np.array([10000, 40000]) * u.K
+    >>> time = np.array([0, 300]) * u.s
+    >>> dt = 0.25 * u.s
+
+    The initial conditions can be accessed using the initial attribute.
+
+    >>> sim = NEI(inputs=inputs, abundances=abund, n=n, T_e=T_e, time_input=time, adapt_dt=False, dt=dt)
+
+    After having inputted all of the necessary information, we can run
+    the simulation.
+
+    >>> results = sim.simulate()
+
+    The initial results are stored in the ``initial`` attribute.
+
+    >>> sim.initial.ionic_fractions['H']
+    array([0.9, 0.1])
+
+    The final results can be access with the ``final`` attribute.
+
+    >>> sim.final.ionic_fractions['H']
+    array([0.16665179, 0.83334821])
+    >>> sim.final.ionic_fractions['He']
+    array([0.88685261, 0.11218358, 0.00096381])
+    >>> sim.final.T_e
+    <Quantity 40000. K>
+
+    Both ``initial`` and ``final`` are instances of the ``IonizationStates``
+    class.
+
+    Notes
+    -----
+    The ionization and recombination rates are from Chianti version
+    8.7.  These rates include radiative and dielectronic recombination.
+    Photoionization is not included.
+    """
```

Comment:
Put in a reference to Shen et al. (2015) Astronomy & Computing, 12, 1, "A Lagrangian scheme for time-dependent ionization simulations of astrophysical plasmas"...in particular eqs 1 & 2

Reply:
Wait, how is it Lagrangian...? This looked rather, er, space-free...

Reply:
In practice, this technique would be used following a parcel of plasma, for example as it moves away from the Sun or passes through a supernova remnant shock wave.  Another example would be taking the output of a numerical simulation and tracing the temperature and density of a parcel of plasma as a function of time.  

In any case, it'll be necessary to clarify this in the forthcoming narrative documentation and be careful about language.  I'll probably avoid the terms "Lagrangian" and "Eulerian" because it always takes me 17 seconds to remember which is which.